### ISSUE TITLE
ModelName requirement in API calls

### ISSUE THREAD

**Issue by ukclivecox** [kserve/kserve#144]
When creating KFservers such as xgboost the `modelName` is taken to be `metdata.Name`. The user then needs to specify this twice in an API call. For example, in the [xgboost sample](https://github.com/kubeflow/kfserving/tree/master/docs/samples/xgboost):

```
MODEL_NAME=xgboost-iris
curl -v -H "Host: xgboost-iris.default.svc.cluster.local" http://$CLUSTER_IP/models/$MODEL_NAME:predict -d $INPUT_PATH
```

 * Should we keep this requirement or allow users to just curl their endpoint and behind the scenes add the `modelName` to the path as we route the request?, e.g. 

```
curl -v -H "Host: xgboost-iris.default.svc.cluster.local" http://$CLUSTER_IP/predict 
```

 * If users can specify the `modelName` then they could try other model names that might happen to be loaded on that server which maybe not what we want.

**Comment by yuzisun** [comment]
I think that routing path is inherited from tfserving, it is for the use case to serve multiple models. If we are keeping KFService as single model serving unit to preserve canary/autoscale features, then `http://$CLUSTER_IP/predict` makes sense

**Comment by ellis-bigelow** [comment]
Yes this sounds good to me. Sounds like we'll need to enable this in the executor or perhaps with an istio virtualservice?

@cliveseldon , does the Seldon protocol specify path?

**Comment by cliveseldon** [comment]
Seldon does specify paths like `/api/v0.1/predictions` but for the kfserver implementation we have conformed to the tensorflow paths that allow model specification.

As the executor is optional I assume an istio VirtualService is best?

**Comment by ellis-bigelow** [comment]
It's not super clear to me how we support this.
                                          _ Default Configuration
PathRewriteVS -> Route /
                                         \_ Canary Configuration

The problem is that this breaks pinned mode. The calling pattern will be different if calling a canary directly.

It's not clear to me if we can replace a Route with an Istio VirtualService. Is the route doing the scale to zero behavior? @yuzisun might know.

Another option is to have a "default" model in the model server that is accessible at HOST:PORT:predict. The problem with this is that it breaks multi model servers.

The executor is cleaner, but has a little overhead.

**Comment by yuzisun** [comment]
ye the route reconciles underlying virtual service to point to activator when scaling down to zero, also it manages the traffic shifting from old revision to new when default/canary configuration is modified. Maybe we should raise this issue to knative ?

**Comment by ellis-bigelow** [comment]
Honestly, I think this is a pretty acceptable quirk of the system. Can we punt on this?

/tag v1beta1
/tag data-plane
